client/packages/core/src/schemaTypes.ts

```diff
@@ -277,19 +277,28 @@ type OptionalKeys<Attrs extends AttrsDefs> = {
  *   - Required keys => `key: ValueType`
  *   - Optional keys => `key?: ValueType`
  */
-type MappedAttrs<Attrs extends AttrsDefs> = {
+type MappedAttrs<Attrs extends AttrsDefs, UseDates extends boolean> = {
   [K in RequiredKeys<Attrs>]: Attrs[K] extends DataAttrDef<infer V, any, any>
-    ? V
+    ? V extends Date
```

Comment:
What will happen here if the user specified Date in their asType, but they don't pass `useDateObjects` e.g.:

```js
i.schema({
  someEntity: i.entity({
    date: i.date()
  }).asType({date: Date})
})
```